drivers/flash/spi_flash_at45.c

```diff
@@ -534,6 +557,26 @@ static int spi_flash_at45_init(const struct device *dev)
 		return -ENODEV;
 	}
 
+	if (dev_config->reset_gpio) {
+		if (gpio_pin_configure(dev_config->reset_gpio,
+					dev_config->reset_pin,
+					GPIO_OUTPUT_ACTIVE | dev_config->reset_dt_flags)) {
+			LOG_ERR("Couldn't configure reset pin");
+			return -ENODEV;
+		}
+		gpio_pin_set(dev_config->reset_gpio, dev_config->reset_pin, 0);
+	}
+
+	if (dev_config->wp_gpio) {
+		if (gpio_pin_configure(dev_config->wp_gpio,
+					dev_config->wp_pin,
+					GPIO_OUTPUT_ACTIVE | dev_config->wp_dt_flags)) {
+			LOG_ERR("Couldn't configure write protect pin");
+			return -ENODEV;
+		}
+		gpio_pin_set(dev_config->wp_gpio, dev_config->wp_pin, 0);
```

Comment:
Is this intentional? Wouldn't it be better to keep the WP pin active and deactivate it only when a write or erase is to be performed?